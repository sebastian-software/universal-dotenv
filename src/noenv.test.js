const savedProcessEnv = { ...process.env }
afterAll(() => {
  process.env = savedProcessEnv
})

/* eslint-disable global-require, @typescript-eslint/no-var-requires */
// Clear pre-existing NODE_ENV variable
delete process.env.NODE_ENV

const snapshotOpts = {
  APP_ROOT: expect.any(String),
  APP_SOURCE: expect.any(String)
}

/* eslint-disable import/no-commonjs */
// We can't use ESM when relying on the fact the the env from the top is correctly respected.
const api = require("..")

api.init()

test("Defaults to development for NODE_ENV", () => {
  const { raw, stringified, webpack } = api.getEnvironment()
  expect(raw).toMatchSnapshot(snapshotOpts)
  expect(stringified).toMatchSnapshot(snapshotOpts)
  expect(webpack).toBeDefined()
})
