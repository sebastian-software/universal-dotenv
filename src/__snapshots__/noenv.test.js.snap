// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Defaults to development for NODE_ENV 1`] = `
Object {
  "APP_DEVELOPMENT": true,
  "APP_ROOT": Any<String>,
  "APP_SOURCE": Any<String>,
  "APP_TEST_OVERRIDE": "original",
  "APP_TEST_PATH": "one/two/three.js",
  "NODE_ENV": "development",
}
`;

exports[`Defaults to development for NODE_ENV 2`] = `
Object {
  "APP_DEVELOPMENT": "true",
  "APP_ROOT": Any<String>,
  "APP_SOURCE": Any<String>,
  "APP_TEST_OVERRIDE": "\\"original\\"",
  "APP_TEST_PATH": "\\"one/two/three.js\\"",
  "NODE_ENV": "\\"development\\"",
}
`;
